py/map.c

```diff
@@ -136,6 +156,17 @@ mp_map_elem_t *mp_map_lookup(mp_map_t *map, mp_obj_t index, mp_map_lookup_kind_t
     // If the map is a fixed array then we must only be called for a lookup
     assert(!map->is_fixed || lookup_kind == MP_MAP_LOOKUP);
 
+    #if MICROPY_OPT_MAP_LOOKUP_CACHE
+    if (lookup_kind == MP_MAP_LOOKUP && map->alloc) {
```

Comment:
I think this can apply just as well to `MP_MAP_LOOKUP_ADD_IF_NOT_FOUND`.

Reply:
Making it `lookup_kind != MP_MAP_LOOKUP_REMOVE_IF_FOUND` gives a nice little boost.

Reply:
Yep, good thinking. On PYBV11:

```
$ ./run-perfbench.py -s ~/3.cache-map-pybv11 ~/3.cache-map-add-pybv11 
diff of scores (higher is better)
N=100 M=100              /home/jimmo/3.cache-map-pybv11 -> /home/jimmo/3.cache-map-add-pybv11         diff      diff% (error%)
bm_asyncio.py                113.73 ->     113.74 :      +0.01 =  +0.009% (+/-0.01%)
bm_chaos.py                  344.82 ->     343.50 :      -1.32 =  -0.383% (+/-0.00%)
bm_fannkuch.py                78.62 ->      78.62 :      +0.00 =  +0.000% (+/-0.00%)
bm_fft.py                   2593.26 ->    2593.13 :      -0.13 =  -0.005% (+/-0.00%)
bm_float.py                 5460.45 ->    5667.22 :    +206.77 =  +3.787% (+/-0.01%)
bm_hexiom.py                  42.78 ->      42.58 :      -0.20 =  -0.468% (+/-0.00%)
bm_nqueens.py               4467.91 ->    4464.29 :      -3.62 =  -0.081% (+/-0.00%)
bm_pidigits.py               648.37 ->     649.28 :      +0.91 =  +0.140% (+/-0.41%)
misc_aes.py                  426.08 ->     425.96 :      -0.12 =  -0.028% (+/-0.01%)
misc_mandel.py              3621.79 ->    3620.72 :      -1.07 =  -0.030% (+/-0.01%)
misc_pystone.py             2289.22 ->    2368.03 :     +78.81 =  +3.443% (+/-0.01%)
misc_raytrace.py             357.00 ->     354.92 :      -2.08 =  -0.583% (+/-0.01%)
```